# Content:
1. Improting Library
2. Loading Data
3. Data Exploration
4. Data Cleaning<br>
4.1. Handling missing values<br>
4.2. Removing Duplicates<br>
4.3. Data Transformation
5. Exporting Cleaned Data

# 1. Importing Library

In [2]:
import pandas as pd
import numpy as np
import os
import re

# 2. Loading Data

In [3]:
# creat path to dataset: 
path = r'/Users/tuyetnguyen/Desktop/Career Foundry/Achievement 6/09-2023 Airbnb Listings and Neighborhoods Analysis'

In [4]:
# import dataset:
df = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'listings.csv'))

# 03 Data Exploration

In [5]:
# View first few rows of the dataset:
df.head()

id                                      listing_url  \
0            13226114            https://www.airbnb.com/rooms/13226114   
1            12518952            https://www.airbnb.com/rooms/12518952   
2  521597880867717063  https://www.airbnb.com/rooms/521597880867717063   
3            17889172            https://www.airbnb.com/rooms/17889172   
4            15917796            https://www.airbnb.com/rooms/15917796   

        scrape_id last_scraped       source  \
0  20230624173239   2023-06-25  city scrape   
1  20230624173239   2023-06-25  city scrape   
2  20230624173239   2023-06-24  city scrape   
3  20230624173239   2023-06-24  city scrape   
4  20230624173239   2023-06-24  city scrape   

                                                name  \
0  Home in Seattle · ★4.79 · 2 bedrooms · 4 beds ...   
1  Guest suite in Seattle · ★5.0 · 2 bedrooms · 6...   
2    Serviced apartment in Seattle · Studio · 1 bath   
3  Rental unit in Seattle · ★5.0 · 1 bedroom · 1 ...   
4  Home in Seattle · ★4.60 · Studio · 3 beds · 1 ...   

                                         description  \
0  Explore Columbia City's  lively culinary, bar ...   
1  Newly painted, beautiful,  bright and wonderfu...   
2  Centrally located in the Adams neighborhood of...   
3  Mid-century apartment well preserved with rece...   
4  The Carriage House is close to Eastlake Union,...   

                               neighborhood_overview  \
0  Columbia City's diverse restaurants and locall...   
1  Many good restaurants and cafés, bathhouse the...   
2                                                NaN   
3  Central Wallingford location. Close to Stone W...   
4  The neighborhood contains a mixture of residen...   

                                         picture_url   host_id  ...  \
0  https://a0.muscache.com/pictures/miso/Hosting-...   1884549  ...   
1  https://a0.muscache.com/pictures/9ff506f6-8927...  12677600  ...   
2  https://a0.muscache.com/pictures/prohost-api/H...  48005494  ...   
3  https://a0.muscache.com/pictures/706ab66c-9cda...  66909032  ...   
4  https://a0.muscache.com/pictures/d56dc8ce-e594...  38021932  ...   

  review_scores_communication review_scores_location review_scores_value  \
0                        5.00                   4.61                4.65   
1                        5.00                   5.00                4.83   
2                        5.00                   3.50                5.00   
3                        5.00                   4.96                4.89   
4                        4.77                   4.93                4.33   

              license instant_bookable calculated_host_listings_count  \
0  STR-OPLI-19-000171                f                              1   
1  STR-OPLI-19-002061                f                              1   
2                 NaN                f                             36   
3  STR-OPLI-19-000865                f                              4   
4  STR-OPLI-19-003016                f                              1   

  calculated_host_listings_count_entire_homes  \
0                                           1   
1                                           1   
2                                          36   
3                                           4   
4                                           1   

  calculated_host_listings_count_private_rooms  \
0                                            0   
1                                            0   
2                                            0   
3                                            0   
4                                            0   

  calculated_host_listings_count_shared_rooms reviews_per_month  
0                                           0              0.28  
1                                           0              0.71  
2                                           0              0.44  
3                                           0              0.38  
4                              

In [6]:
# Check data types and non-null counts
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6636 entries, 0 to 6635
Data columns (total 75 columns):
 #   Column                                        Non-Null Count  Dtype  
---  ------                                        --------------  -----  
 0   id                                            6636 non-null   int64  
 1   listing_url                                   6636 non-null   object 
 2   scrape_id                                     6636 non-null   int64  
 3   last_scraped                                  6636 non-null   object 
 4   source                                        6636 non-null   object 
 5   name                                          6636 non-null   object 
 6   description                                   6630 non-null   object 
 7   neighborhood_overview                         4859 non-null   object 
 8   picture_url                                   6636 non-null   object 
 9   host_id                                       6636 non-null   i

In [7]:
# Check data types and non-null counts
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6636 entries, 0 to 6635
Data columns (total 75 columns):
 #   Column                                        Non-Null Count  Dtype  
---  ------                                        --------------  -----  
 0   id                                            6636 non-null   int64  
 1   listing_url                                   6636 non-null   object 
 2   scrape_id                                     6636 non-null   int64  
 3   last_scraped                                  6636 non-null   object 
 4   source                                        6636 non-null   object 
 5   name                                          6636 non-null   object 
 6   description                                   6630 non-null   object 
 7   neighborhood_overview                         4859 non-null   object 
 8   picture_url                                   6636 non-null   object 
 9   host_id                                       6636 non-null   i

In [8]:
# Create a list of columns that includes the pertinent data necessary for the analysis
pertinent_columns = [
    'id',
    'room_type',
    'accommodates',
    'bedrooms',
    'bathrooms_text',
    'amenities',
    'price',
    'minimum_nights',
    'maximum_nights',
    'number_of_reviews',
    'review_scores_rating',
    'review_scores_accuracy',
    'review_scores_cleanliness',
    'review_scores_checkin',
    'review_scores_communication',
    'review_scores_location',
    'review_scores_value',
    'neighbourhood_cleansed',
    'latitude',
    'longitude',
]

In [9]:
# Create a new DataFrame with the same name 'df_pertinent'
df = df.loc[:, pertinent_columns]

In [10]:
# Check the new DataFrame
df.head()

id        room_type  accommodates  bedrooms bathrooms_text  \
0            13226114  Entire home/apt             5       2.0        2 baths   
1            12518952  Entire home/apt             4       2.0         1 bath   
2  521597880867717063  Entire home/apt             2       NaN         1 bath   
3            17889172  Entire home/apt             2       1.0         1 bath   
4            15917796  Entire home/apt             4       NaN         1 bath   

                                           amenities    price  minimum_nights  \
0  ["Private patio or balcony", "Ceiling fan", "B...  $240.00               4   
1  ["45\" HDTV with Netflix, HBO Max, Hulu, Amazo...  $200.00               3   
2  ["Dryer \u2013\u00a0In unit", "Carbon monoxide...   $81.00              30   
3  ["Baking sheet", "Carbon monoxide alarm", "Dis...  $125.00              30   
4  ["Free driveway parking on premises \u2013 1 s...  $128.00              30   

   maximum_nights  number_of_reviews  review_scores_rating  \
0             200                 24                  4.79   
1            1125                 60                  5.00   
2             731                  2                  5.00   
3             150                 28                  5.00   
4            1125                 31                  4.60   

   review_scores_accuracy  review_scores_cleanliness  review_scores_checkin  \
0                    4.92                       4.83                   5.00   
1                    4.98                       4.98                   4.98   
2                    5.00                       5.00                   5.00   
3                    4.96                       4.96                   5.00   
4                    4.77                       4.53                   4.87   

   review_scores_communication  review_scores_location  review_scores_value  \
0                         5.00                    4.61                 4.65   
1                         5.00                    5.00                 4.83   
2                         5.00                    3.50                 5.00   
3                         5.00                    4.96                 4.89   
4                         4.77                    4.93                 4.33   

  neighbourhood_cleansed  latitude  longitude  
0          Columbia City  47.56555 -122.29385  
1             Green Lake  47.68243 -122.33086  
2                  Adams  47.66646 -122.37650  
3            Wallingford  47.65480 -122.34042  
4               Montlake  47.64017 -122.32271

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6636 entries, 0 to 6635
Data columns (total 20 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   id                           6636 non-null   int64  
 1   room_type                    6636 non-null   object 
 2   accommodates                 6636 non-null   int64  
 3   bedrooms                     5060 non-null   float64
 4   bathrooms_text               6635 non-null   object 
 5   amenities                    6636 non-null   object 
 6   price                        6636 non-null   object 
 7   minimum_nights               6636 non-null   int64  
 8   maximum_nights               6636 non-null   int64  
 9   number_of_reviews            6636 non-null   int64  
 10  review_scores_rating         5562 non-null   float64
 11  review_scores_accuracy       5559 non-null   float64
 12  review_scores_cleanliness    5559 non-null   float64
 13  review_scores_chec

In [12]:
# rename 'bathrooms_text' to 'bathrooms' before changing its datatype to numeric
df.rename(columns={'bathrooms_text':'bathrooms', 'neighbourhood_cleansed':'neighborhood'}, inplace=True)

In [13]:
# Convert 'bathrooms' into numeric value
df['bathrooms'] = df['bathrooms'].str.extract('(\d+)').astype(float)

In [14]:
# Check 'bathrooms' type:
print(df['bathrooms'].dtypes)

float64


In [15]:
# Obtain a summary statistics for the numeric columns:
df.describe()

id  accommodates     bedrooms    bathrooms  minimum_nights  \
count  6.636000e+03   6636.000000  5060.000000  6622.000000     6636.000000   
mean   3.075627e+17      4.009192     1.930435     1.352008       11.940024   
std    3.715633e+17      2.552078     1.129622     0.733935       19.000116   
min    6.606000e+03      0.000000     1.000000     0.000000        1.000000   
25%    2.684482e+07      2.000000     1.000000     1.000000        2.000000   
50%    5.168138e+07      4.000000     2.000000     1.000000        2.000000   
75%    7.065462e+17      5.000000     2.000000     2.000000       30.000000   
max    9.198184e+17     16.000000    13.000000    16.000000      365.000000   

       maximum_nights  number_of_reviews  review_scores_rating  \
count     6636.000000        6636.000000           5562.000000   
mean       514.682037          64.055154              4.786535   
std        490.996284         103.263320              0.391729   
min          1.000000           0.000000              0.000000   
25%         31.000000           2.000000              4.750000   
50%        365.000000          23.000000              4.890000   
75%       1125.000000          80.000000              4.980000   
max      10000.000000        1281.000000              5.000000   

       review_scores_accuracy  review_scores_cleanliness  \
count             5559.000000                5559.000000   
mean                 4.828030                   4.807971   
std                  0.345872                   0.348809   
min                  1.000000                   1.000000   
25%                  4.810000                   4.770000   
50%                  4.930000                   4.920000   
75%                  5.000000                   5.000000   
max                  5.000000                   5.000000   

       review_scores_checkin  review_scores_communication  \
count            5559.000000                  5559.000000   
mean                4.873056                     4.864614   
std                 0.303903                     0.350205   
min                 1.000000                     1.000000   
25%                 4.880000                     4.880000   
50%                 4.960000                     4.970000   
75%                 5.000000                     5.000000   
max                 5.000000                     5.000000   

       review_scores_location  review_scores_value     latitude    longitude  
count             5559.000000          5559.000000  6636.000000  6636.000000  
mean                 4.833412             4.710536    47.626700  -122.334122  
std                  0.293526             0.391589     0.049349     0.033042  
min                  1.000000             1.000000    47.495874  -122.419050  
25%                  4.800000             4.660000    47.601959  -122.356382  
50%                  4.910000             4.810000    47.623265  -122.333007  
75%                  4.980000             4.910000    47.664345  -122.311175  
max                  5.000000             5.000000    47.734010  -122.240870

# 4. Data Cleaning 

## 4.1. Handling missing values

In [16]:
# Check for missing values in the entire DataFrame
missing_data = df.isnull().sum()
print(missing_data)

id                                0
room_type                         0
accommodates                      0
bedrooms                       1576
bathrooms                        14
amenities                         0
price                             0
minimum_nights                    0
maximum_nights                    0
number_of_reviews                 0
review_scores_rating           1074
review_scores_accuracy         1077
review_scores_cleanliness      1077
review_scores_checkin          1077
review_scores_communication    1077
review_scores_location         1077
review_scores_value            1077
neighborhood                      0
latitude                          0
longitude                         0
dtype: int64


<p><b>'bedrooms' column:</b><br>
Upon examining the dataset, it appears that most entries with missing values correspond to private rooms. Therefore, the analysis has decided to fill in the blanks with a value of 1.</p>

In [17]:
df['bedrooms'].fillna(1, inplace=True)

<p><b>'bathrooms' column:</b><br>
# Given that there are only a few entries with missing values (14 out of 6636), 
# The analyst has opted to remove these rows, as the impact on the analysis is minimal.</p>


In [18]:
df.dropna(subset=['bathrooms'], inplace=True)

<b>'reviews_scores_...' columns:</b><br>
Upon dataset examination, most missing values are primarily associated with listings with 0 availability and the analyst cannot reasonably impute these missing values. Hence, the analyst decides to exclude them to improve the quality of your analysis.</p>





In [19]:
df.dropna(subset=['review_scores_accuracy'], inplace=True)

In [20]:
# check the DataFrame:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5547 entries, 0 to 6635
Data columns (total 20 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   id                           5547 non-null   int64  
 1   room_type                    5547 non-null   object 
 2   accommodates                 5547 non-null   int64  
 3   bedrooms                     5547 non-null   float64
 4   bathrooms                    5547 non-null   float64
 5   amenities                    5547 non-null   object 
 6   price                        5547 non-null   object 
 7   minimum_nights               5547 non-null   int64  
 8   maximum_nights               5547 non-null   int64  
 9   number_of_reviews            5547 non-null   int64  
 10  review_scores_rating         5547 non-null   float64
 11  review_scores_accuracy       5547 non-null   float64
 12  review_scores_cleanliness    5547 non-null   float64
 13  review_scores_chec

In [21]:
# check the DataFrame:
df.head()

id        room_type  accommodates  bedrooms  bathrooms  \
0            13226114  Entire home/apt             5       2.0        2.0   
1            12518952  Entire home/apt             4       2.0        1.0   
2  521597880867717063  Entire home/apt             2       1.0        1.0   
3            17889172  Entire home/apt             2       1.0        1.0   
4            15917796  Entire home/apt             4       1.0        1.0   

                                           amenities    price  minimum_nights  \
0  ["Private patio or balcony", "Ceiling fan", "B...  $240.00               4   
1  ["45\" HDTV with Netflix, HBO Max, Hulu, Amazo...  $200.00               3   
2  ["Dryer \u2013\u00a0In unit", "Carbon monoxide...   $81.00              30   
3  ["Baking sheet", "Carbon monoxide alarm", "Dis...  $125.00              30   
4  ["Free driveway parking on premises \u2013 1 s...  $128.00              30   

   maximum_nights  number_of_reviews  review_scores_rating  \
0             200                 24                  4.79   
1            1125                 60                  5.00   
2             731                  2                  5.00   
3             150                 28                  5.00   
4            1125                 31                  4.60   

   review_scores_accuracy  review_scores_cleanliness  review_scores_checkin  \
0                    4.92                       4.83                   5.00   
1                    4.98                       4.98                   4.98   
2                    5.00                       5.00                   5.00   
3                    4.96                       4.96                   5.00   
4                    4.77                       4.53                   4.87   

   review_scores_communication  review_scores_location  review_scores_value  \
0                         5.00                    4.61                 4.65   
1                         5.00                    5.00                 4.83   
2                         5.00                    3.50                 5.00   
3                         5.00                    4.96                 4.89   
4                         4.77                    4.93                 4.33   

    neighborhood  latitude  longitude  
0  Columbia City  47.56555 -122.29385  
1     Green Lake  47.68243 -122.33086  
2          Adams  47.66646 -122.37650  
3    Wallingford  47.65480 -122.34042  
4       Montlake  47.64017 -122.32271

## 4.2. Removing duplicates

In [22]:
# Remove duplicate rows:
df.drop_duplicates(inplace=True)

## 4.3. Data Transformation


<p><b>'prices' Column</b><br>

In [23]:
# Remove the '$' sign and convert the 'price' column to float
df['price'] = df['price'].str.replace('[\$,]', '', regex=True).astype(float)

<p><b>'amenities' Column</b><br>
The 'amenities' column includes a textual list of provided amenities for each listing. To enhance the data's usability for analysis, the analyst has decided to split the text based on the delimiter (comma) and create new binary columns for each amenity. </p>

In [24]:
# Parse amenities using a comma delimiter
# Remove curly braces and quotes
df['amenities'] = df['amenities'].str.replace('["]', '', regex=True)  

In [25]:
# Split the modified 'amenities' column by commas ,. 
df['amenities'] = df['amenities'].str.split(',') 

In [26]:
# Join the string using a pipe symbol ('|') 
# then converts this string into a DataFrame of binary columns.
df_amenities = df['amenities'].str.join('|').str.get_dummies()

In [27]:
# view the df_amenities DataFrame
df_amenities.head()

Beko stainless steel oven    Paraben & Sulfate Free body soap  \
0                            0                                   0   
1                            0                                   0   
2                            0                                   0   
3                            0                                   0   
4                            0                                   0   

     Paraben & Sulfate Free conditioner    Paraben & Sulfate Free shampoo  \
0                                     0                                 0   
1                                     0                                 0   
2                                     0                                 0   
3                                     0                                 0   
4                                     0                                 0   

    & Planet shampoo  \
0                  0   
1                  0   
2                  0   
3                  0   
4                  0   

    1 Harman Kardon Onyx Studio 4 wireless bluetooth speakers Bluetooth sound system  \
0                                                  0                                   
1                                                  0                                   
2                                                  0                                   
3                                                  0                                   
4                                                  0                                   

    1-burner induction stovetop induction stove   100% Vegan  \
0                                             0            0   
1                                             0            0   
2                                             0            0   
3                                             0            0   
4                                             0            0   

    100% Vegan & Cruelty-Free   100% Vegan & Cruelty-Free shampoo  ...  \
0                           0                                   0  ...   
1                           0                                   0  ...   
2                           0                                   0  ...   
3                           0                                   0  ...   
4                           0                                   0  ...   

   [TV with standard cable  [The Unscented Company (made in Canada) shampoo  \
0                        0                                                0   
1                        0                                                0   
2                        0                                                0   
3                        0                                                0   
4                        0                                                0   

   [Trader Joe's shampoo  [Trader joes conditioner  [Washer  \
0                      0                         0        0   
1                      0                         0        0   
2                      0                         0        0   
3                      0                         0        0   
4                      0                         0        0   

   [Whirlpool stainless steel oven  [White color electric stove  [Wifi  \
0                                0                            0      0   
1                                0                            0      0   
2                                0                            0      0   
3                                0                            0      0   
4                                0                            0      0   

   [Window AC unit  electric  
0                0         0  
1                0         0  
2                0         0  
3                0         0  
4                0         0  

[5 rows x 2926 columns]

In [28]:
df_amenities.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5547 entries, 0 to 6635
Columns: 2926 entries,   Beko stainless steel oven to electric
dtypes: int64(2926)
memory usage: 123.9 MB


In [29]:
# Calculate the total number of amenities for each listing 
df['total_amenities'] = df_amenities.sum(axis=1)

In [30]:
# Check the modified 'listings' DataFrame
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5547 entries, 0 to 6635
Data columns (total 21 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   id                           5547 non-null   int64  
 1   room_type                    5547 non-null   object 
 2   accommodates                 5547 non-null   int64  
 3   bedrooms                     5547 non-null   float64
 4   bathrooms                    5547 non-null   float64
 5   amenities                    5547 non-null   object 
 6   price                        5547 non-null   float64
 7   minimum_nights               5547 non-null   int64  
 8   maximum_nights               5547 non-null   int64  
 9   number_of_reviews            5547 non-null   int64  
 10  review_scores_rating         5547 non-null   float64
 11  review_scores_accuracy       5547 non-null   float64
 12  review_scores_cleanliness    5547 non-null   float64
 13  review_scores_chec

In [31]:
df.head()

id        room_type  accommodates  bedrooms  bathrooms  \
0            13226114  Entire home/apt             5       2.0        2.0   
1            12518952  Entire home/apt             4       2.0        1.0   
2  521597880867717063  Entire home/apt             2       1.0        1.0   
3            17889172  Entire home/apt             2       1.0        1.0   
4            15917796  Entire home/apt             4       1.0        1.0   

                                           amenities  price  minimum_nights  \
0  [[Private patio or balcony,  Ceiling fan,  Bak...  240.0               4   
1  [[45\ HDTV with Netflix,  HBO Max,  Hulu,  Ama...  200.0               3   
2  [[Dryer \u2013\u00a0In unit,  Carbon monoxide ...   81.0              30   
3  [[Baking sheet,  Carbon monoxide alarm,  Dishw...  125.0              30   
4  [[Free driveway parking on premises \u2013 1 s...  128.0              30   

   maximum_nights  number_of_reviews  ...  review_scores_accuracy  \
0             200                 24  ...                    4.92   
1            1125                 60  ...                    4.98   
2             731                  2  ...                    5.00   
3             150                 28  ...                    4.96   
4            1125                 31  ...                    4.77   

   review_scores_cleanliness  review_scores_checkin  \
0                       4.83                   5.00   
1                       4.98                   4.98   
2                       5.00                   5.00   
3                       4.96                   5.00   
4                       4.53                   4.87   

   review_scores_communication  review_scores_location  review_scores_value  \
0                         5.00                    4.61                 4.65   
1                         5.00                    5.00                 4.83   
2                         5.00                    3.50                 5.00   
3                         5.00                    4.96                 4.89   
4                         4.77                    4.93                 4.33   

    neighborhood  latitude  longitude  total_amenities  
0  Columbia City  47.56555 -122.29385               67  
1     Green Lake  47.68243 -122.33086               68  
2          Adams  47.66646 -122.37650               30  
3    Wallingford  47.65480 -122.34042               35  
4       Montlake  47.64017 -122.32271               21  

[5 rows x 21 columns]

In [32]:
# Drop the unnamed column 
df = df.drop(df.columns[0], axis = 1)

In [33]:
df.head()

room_type  accommodates  bedrooms  bathrooms  \
0  Entire home/apt             5       2.0        2.0   
1  Entire home/apt             4       2.0        1.0   
2  Entire home/apt             2       1.0        1.0   
3  Entire home/apt             2       1.0        1.0   
4  Entire home/apt             4       1.0        1.0   

                                           amenities  price  minimum_nights  \
0  [[Private patio or balcony,  Ceiling fan,  Bak...  240.0               4   
1  [[45\ HDTV with Netflix,  HBO Max,  Hulu,  Ama...  200.0               3   
2  [[Dryer \u2013\u00a0In unit,  Carbon monoxide ...   81.0              30   
3  [[Baking sheet,  Carbon monoxide alarm,  Dishw...  125.0              30   
4  [[Free driveway parking on premises \u2013 1 s...  128.0              30   

   maximum_nights  number_of_reviews  review_scores_rating  \
0             200                 24                  4.79   
1            1125                 60                  5.00   
2             731                  2                  5.00   
3             150                 28                  5.00   
4            1125                 31                  4.60   

   review_scores_accuracy  review_scores_cleanliness  review_scores_checkin  \
0                    4.92                       4.83                   5.00   
1                    4.98                       4.98                   4.98   
2                    5.00                       5.00                   5.00   
3                    4.96                       4.96                   5.00   
4                    4.77                       4.53                   4.87   

   review_scores_communication  review_scores_location  review_scores_value  \
0                         5.00                    4.61                 4.65   
1                         5.00                    5.00                 4.83   
2                         5.00                    3.50                 5.00   
3                         5.00                    4.96                 4.89   
4                         4.77                    4.93                 4.33   

    neighborhood  latitude  longitude  total_amenities  
0  Columbia City  47.56555 -122.29385               67  
1     Green Lake  47.68243 -122.33086               68  
2          Adams  47.66646 -122.37650               30  
3    Wallingford  47.65480 -122.34042               35  
4       Montlake  47.64017 -122.32271               21

In [34]:
df.columns

Index(['room_type', 'accommodates', 'bedrooms', 'bathrooms', 'amenities',
       'price', 'minimum_nights', 'maximum_nights', 'number_of_reviews',
       'review_scores_rating', 'review_scores_accuracy',
       'review_scores_cleanliness', 'review_scores_checkin',
       'review_scores_communication', 'review_scores_location',
       'review_scores_value', 'neighborhood', 'latitude', 'longitude',
       'total_amenities'],
      dtype='object')

# 5. Exporting Cleaned Data

In [35]:
df.to_csv(os.path.join(path, '02 Data','Prepared Data', 'listings_cleaned.csv'))